### PR TITLE
Refactor table and simplify calls

### PR DESCRIPTION
**Pull request by gumb0**
Pulled out of #615

### CODE REVIEW

File: lib/fizzy/instantiate.hpp [hunk 1]
```diff
@@ -28,7 +28,18 @@ struct ExternalFunction
...
-using table_elements = std::vector<std::optional<ExternalFunction>>;
+// Table element, which references a function in any instance.
+struct TableElement
+{
+    Instance* instance;
+    FuncIdx func_idx;
+    // This pointer is empty most of the time and is used only to keep instance alive in one edge
+    // case, when start function traps, but instantiate has already modified some elements of a
+    // shared (imported) table.
+    std::shared_ptr<Instance> shared_instance;
```
Comment (gumb0) [inline lib/fizzy/instantiate.hpp:39]: This is a simple way to keep instance alive as long as there is function using it in some table. This was previously achieved by capturing this `shared_ptr<Instance>` in `std::function` of table element.

Now this looks like an overhead included in each table element every time, but needed only in weird edge case.
But on the other hand, previously there was `std::function`'s dynamic allocation overhead for each table element all the time, and now this `shared_ptr` stays empty in normal cases, so this theoretically should be overall more efficient than before.

**Comment by gumb0** [comment]
Clang10 with LTO
```
fizzy/parse/blake2b_mean                                            +0.0052         +0.0058            45            46            45            46
fizzy/instantiate/blake2b_mean                                      +0.0090         +0.0087            52            52            52            52
fizzy/execute/blake2b/512_bytes_rounds_1_mean                       +0.1245         +0.1239           173           195           173           195
fizzy/execute/blake2b/512_bytes_rounds_16_mean                      +0.1241         +0.1255          2559          2877          2555          2875
fizzy/parse/ecpairing_mean                                          -0.0286         -0.0279          2689          2612          2686          2611
fizzy/instantiate/ecpairing_mean                                    -0.0297         -0.0291          2789          2707          2788          2707
fizzy/execute/ecpairing/onepoint_mean                               +0.0060         +0.0063        875511        880740        875390        880863
fizzy/parse/keccak256_mean                                          +0.0231         +0.0215            83            85            83            85
fizzy/instantiate/keccak256_mean                                    +0.0263         +0.0254            89            91            89            91
fizzy/execute/keccak256/512_bytes_rounds_1_mean                     +0.0108         +0.0103           232           235           232           235
fizzy/execute/keccak256/512_bytes_rounds_16_mean                    -0.0030         -0.0035          3352          3342          3354          3342
fizzy/parse/memset_mean                                             -0.0228         -0.0199            11            11            11            11
fizzy/instantiate/memset_mean                                       -0.0093         -0.0079            17            17            17            17
fizzy/execute/memset/256_bytes_mean                                 +0.0182         +0.0190            14            14            14            14
fizzy/execute/memset/60000_bytes_mean                               +0.0080         +0.0084          3102          3127          3101          3127
fizzy/parse/mul256_opt0_mean                                        -0.0271         -0.0281            15            14            15            14
fizzy/instantiate/mul256_opt0_mean                                  -0.0126         -0.0134            21            20            21            20
fizzy/execute/mul256_opt0/input1_mean                               -0.0956         -0.0961            65            59            65            59
fizzy/parse/ramanujan_pi_mean                                       +0.0286         +0.0283            49            50            49            50
fizzy/instantiate/ramanujan_pi_mean                                 +0.0323         +0.0321            55            57            55            57
fizzy/execute/ramanujan_pi/33_runs_mean                             -0.0117         -0.0118           233           230           233           230
fizzy/parse/sha1_mean                                               +0.0084         +0.0083            75            76            75            76
fizzy/instantiate/sha1_mean                                         +0.0134         +0.0133            82            83            82            83
fizzy/execute/sha1/512_bytes_rounds_1_mean                          +0.0118         +0.0121           182           184           182           184
fizzy/execute/sha1/512_bytes_rounds_16_mean                         +0.0173         +0.0174          2516          2560          2515          2559
fizzy/parse/sha256_mean                                             +0.0059         +0.0060           127           128           127           128
fizzy/instantiate/sha256_mean                                       +0.0138         +0.0139           134           135           133           135
fizzy/execute/sha256/512_bytes_rounds_1_mean                        -0.0099         -0.0098           177           176           177           176
fizzy/execute/sha256/512_bytes_rounds_16_mean                       -0.0105         -0.0104          2416          2390          2415          2390
fizzy/parse/taylor_pi_mean                                          -0.0181         -0.0181             5             5             5             5
fizzy/instantiate/taylor_pi_mean                                    +0.0386         +0.0389            11            11            11            11
fizzy/execute/taylor_pi/pi_1000000_runs_mean                        +0.0012         +0.0012         87531         87634         87522         87626
fizzy/parse/micro/eli_interpreter_mean                              -0.0140         -0.0140             8             8             8             8
fizzy/instantiate/micro/eli_interpreter_mean                        +0.0071         +0.0071            14            14            14            14
fizzy/execute/micro/eli_interpreter/exec105_mean                    +0.0131         +0.0131             9             9             9             9
fizzy/parse/micro/factorial_mean                                    -0.0194         -0.0195             2             2             2             2
fizzy/instantiate/micro/factorial_mean                              +0.0003         +0.0002             2             2             2             2
fizzy/execute/micro/factorial/20_mean                               +0.0042         +0.0041             1             1             1             1
fizzy/parse/micro/fibonacci_mean                                    +0.0024         +0.0023             2             2             2             2
fizzy/instantiate/micro/fibonacci_mean                              -0.0025         -0.0026             3             3             3             3
fizzy/execute/micro/fibonacci/24_mean                               +0.2125         +0.2124         10846         13151         10846         13149
fizzy/parse/micro/host_adler32_mean                                 -0.0161         -0.0162             3             3             3             3
fizzy/instantiate/micro/host_adler32_mean                           +0.0035         +0.0046             7             7             7             7
fizzy/execute/micro/host_adler32/1_mean                             +0.0283         +0.0293             0             0             0             0
fizzy/execute/micro/host_adler32/1000_mean                          +0.0067         +0.0075            69            69            69            69
fizzy/parse/micro/spinner_mean                                      +0.0078         +0.0084             2             2             2             2
fizzy/instantiate/micro/spinner_mean                                +0.0218         +0.0223             2             2             2             2
fizzy/execute/micro/spinner/1_mean                                  +0.0740         +0.0745             0             0             0             0
fizzy/execute/micro/spinner/1000_mean                               +0.0331         +0.0334            19            19            19            19
fizzy/parse/stress/guido-fuzzer-find-1_mean                         -0.0407         -0.0404           278           267           278           267
fizzy/instantiate/stress/guido-fuzzer-find-1_mean                   -0.0433         -0.0431           326           312           326           312

```

File: lib/fizzy/instantiate.hpp (hunk 1 above)
Comment (chfast) [reply lib/fizzy/instantiate.hpp:39]: This is better now. But I'd still prefer to handle that differently (not necessarily in this PR): either separate `start()` from `instantiate()` or return tuble `Instance*, bool` from `instantiate()`.

File: lib/fizzy/instantiate.hpp [hunk 2]
```diff
@@ -28,7 +28,18 @@ struct ExternalFunction
...
+};
+
+using table_elements = std::vector<std::optional<TableElement>>;
```
Comment (chfast) [inline lib/fizzy/instantiate.hpp:42]: Now having `TableElement`, you can check `element.instance` for empty and skip `std::optional`.

Comment (gumb0) [reply lib/fizzy/instantiate.hpp]: Yeah, goo idea.

File: lib/fizzy/execute.cpp [hunk 3]
```diff
@@ -625,18 +625,18 @@ ExecutionResult execute(Instance& instance, FuncIdx func_idx, const Value* args,
                 goto trap;
 
             const auto called_func = (*instance.table)[elem_idx];
-            if (!called_func.has_value())
+            if (!called_func.instance)
```
Comment (chfast) [inline lib/fizzy/execute.cpp:628]:
```suggestion
            if (!called_func.instance)  // Table element not initialized.
```

File: lib/fizzy/instantiate.hpp [hunk 4]
```diff
@@ -28,7 +28,18 @@ struct ExternalFunction
     FuncType type;
 };
 
-using table_elements = std::vector<std::optional<ExternalFunction>>;
+// Table element, which references a function in any instance.
+struct TableElement
+{
+    Instance* instance;
```
Comment (chfast) [inline lib/fizzy/instantiate.hpp:34]:
```suggestion
    Instance* instance = nullptr;
```

File: lib/fizzy/instantiate.hpp [hunk 5]
```diff
@@ -28,7 +28,18 @@ struct ExternalFunction
...
+    FuncIdx func_idx;
```
Comment (chfast) [inline lib/fizzy/instantiate.hpp:35]:
```suggestion
    FuncIdx func_idx = 0;
```

**Review by chfast — APPROVED** [review]

**Comment by chfast** [comment]
AMD EPYC 7601, GCC10, no LTO

```
fizzy/parse/blake2b_mean                                           -0.0269         -0.0276            51            50            51            50
fizzy/instantiate/blake2b_mean                                     -0.0302         -0.0308            57            55            57            55
fizzy/execute/blake2b/512_bytes_rounds_1_mean                      -0.0362         -0.0368           265           255           265           255
fizzy/execute/blake2b/512_bytes_rounds_16_mean                     -0.0353         -0.0364          4028          3886          4028          3881
fizzy/parse/ecpairing_mean                                         -0.0768         -0.0767          2859          2640          2859          2640
fizzy/instantiate/ecpairing_mean                                   -0.0681         -0.0684          2976          2774          2975          2772
fizzy/execute/ecpairing/onepoint_mean                              +0.0027         +0.0026       1431894       1435826       1431741       1435492
fizzy/parse/keccak256_mean                                         -0.0538         -0.0538            95            90            95            90
fizzy/instantiate/keccak256_mean                                   -0.0430         -0.0430           100            96           100            96
fizzy/execute/keccak256/512_bytes_rounds_1_mean                    +0.0298         +0.0301           295           304           295           304
fizzy/execute/keccak256/512_bytes_rounds_16_mean                   +0.0339         +0.0339          4348          4495          4347          4495
fizzy/parse/memset_mean                                            -0.0325         -0.0326            13            13            13            13
fizzy/instantiate/memset_mean                                      -0.0191         -0.0190            19            18            19            18
fizzy/execute/memset/256_bytes_mean                                +0.0139         +0.0139            21            21            21            21
fizzy/execute/memset/60000_bytes_mean                              +0.0344         +0.0343          4462          4616          4462          4615
fizzy/parse/mul256_opt0_mean                                       -0.0520         -0.0520            18            17            18            17
fizzy/instantiate/mul256_opt0_mean                                 -0.0264         -0.0264            23            22            23            22
fizzy/execute/mul256_opt0/input1_mean                              -0.0815         -0.0816           114           105           114           105
fizzy/parse/ramanujan_pi_mean                                      -0.0232         -0.0232            54            53            54            53
fizzy/instantiate/ramanujan_pi_mean                                -0.0150         -0.0146            59            58            59            58
fizzy/execute/ramanujan_pi/33_runs_mean                            -0.0210         -0.0210           451           441           451           441
fizzy/parse/sha1_mean                                              -0.0307         -0.0303            85            83            85            83
fizzy/instantiate/sha1_mean                                        -0.0215         -0.0214            91            89            91            89
fizzy/execute/sha1/512_bytes_rounds_1_mean                         +0.0727         +0.0715           306           328           306           328
fizzy/execute/sha1/512_bytes_rounds_16_mean                        +0.0881         +0.0827          4283          4660          4282          4637
fizzy/parse/sha256_mean                                            -0.0419         -0.0421           142           136           142           136
fizzy/instantiate/sha256_mean                                      -0.0330         -0.0330           148           143           148           143
fizzy/execute/sha256/512_bytes_rounds_1_mean                       +0.0238         +0.0239           379           388           379           388
fizzy/execute/sha256/512_bytes_rounds_16_mean                      +0.0987         +0.0986          5308          5832          5307          5830
fizzy/parse/taylor_pi_mean                                         +0.0325         +0.0323             6             6             6             6
fizzy/instantiate/taylor_pi_mean                                   +0.0706         +0.0700            11            12            11            12
fizzy/execute/taylor_pi/pi_1000000_runs_mean                       +0.1066         +0.1065        112718        124731        112702        124703
fizzy/parse/micro/eli_interpreter_mean                             -0.0204         -0.0204             8             8             8             8
fizzy/instantiate/micro/eli_interpreter_mean                       -0.0054         -0.0054            14            14            14            14
fizzy/execute/micro/eli_interpreter/exec105_mean                   -0.1249         -0.1249            17            15            17            15
fizzy/parse/micro/factorial_mean                                   -0.0123         -0.0126             2             2             2             2
fizzy/instantiate/micro/factorial_mean                             +0.0484         +0.0483             2             3             2             3
fizzy/execute/micro/factorial/20_mean                              -0.0405         -0.0403             1             1             1             1
fizzy/parse/micro/fibonacci_mean                                   -0.0230         -0.0233             3             3             3             3
fizzy/instantiate/micro/fibonacci_mean                             +0.0043         +0.0042             3             3             3             3
fizzy/execute/micro/fibonacci/24_mean                              -0.0308         -0.0305         12916         12519         12911         12518
fizzy/parse/micro/host_adler32_mean                                -0.0018         -0.0019             3             3             3             3
fizzy/instantiate/micro/host_adler32_mean                          +0.0166         +0.0165             7             7             7             7
fizzy/execute/micro/host_adler32/1_mean                            +0.0333         +0.0330             0             0             0             0
fizzy/execute/micro/host_adler32/1000_mean                         +0.0516         +0.0518            59            62            59            62
fizzy/parse/micro/spinner_mean                                     -0.0090         -0.0093             2             2             2             2
fizzy/instantiate/micro/spinner_mean                               -0.0003         -0.0004             2             2             2             2
fizzy/execute/micro/spinner/1_mean                                 -0.0649         -0.0649             0             0             0             0
fizzy/execute/micro/spinner/1000_mean                              -0.0395         -0.0393            18            18            18            18
fizzy/parse/stress/guido-fuzzer-find-1_mean                        -0.0182         -0.0182           237           233           237           233
fizzy/instantiate/stress/guido-fuzzer-find-1_mean                  -0.0210         -0.0208           280           274           280           274
```

**Comment by chfast** [comment]
AMD EPYC 7601, GCC10, LTO
```
fizzy/parse/blake2b_mean                                           -0.0217         -0.0215            51            50            51            50
fizzy/instantiate/blake2b_mean                                     -0.0175         -0.0175            57            56            57            56
fizzy/execute/blake2b/512_bytes_rounds_1_mean                      -0.0347         -0.0344           291           281           291           281
fizzy/execute/blake2b/512_bytes_rounds_16_mean                     -0.0372         -0.0373          4420          4255          4419          4254
fizzy/parse/ecpairing_mean                                         -0.0084         -0.0082          2730          2707          2729          2707
fizzy/instantiate/ecpairing_mean                                   -0.0387         -0.0388          2936          2823          2936          2822
fizzy/execute/ecpairing/onepoint_mean                              -0.0423         -0.0424       1424255       1363966       1424008       1363597
fizzy/parse/keccak256_mean                                         -0.0092         -0.0090            93            92            93            92
fizzy/instantiate/keccak256_mean                                   -0.0207         -0.0209            99            97            99            97
fizzy/execute/keccak256/512_bytes_rounds_1_mean                    -0.0279         -0.0274           302           294           302           294
fizzy/execute/keccak256/512_bytes_rounds_16_mean                   -0.0279         -0.0282          4452          4328          4452          4326
fizzy/parse/memset_mean                                            +0.0170         +0.0173            13            13            13            13
fizzy/instantiate/memset_mean                                      -0.0126         -0.0127            19            18            19            18
fizzy/execute/memset/256_bytes_mean                                +0.0501         +0.0503            21            22            21            22
fizzy/execute/memset/60000_bytes_mean                              +0.0377         +0.0376          4590          4763          4590          4763
fizzy/parse/mul256_opt0_mean                                       -0.0274         -0.0276            17            17            17            17
fizzy/instantiate/mul256_opt0_mean                                 -0.0257         -0.0255            23            22            23            22
fizzy/execute/mul256_opt0/input1_mean                              -0.0581         -0.0583           117           111           117           111
fizzy/parse/ramanujan_pi_mean                                      -0.0234         -0.0230            53            52            53            52
fizzy/instantiate/ramanujan_pi_mean                                -0.0215         -0.0218            59            58            59            58
fizzy/execute/ramanujan_pi/33_runs_mean                            -0.1367         -0.1352           518           447           517           447
fizzy/parse/sha1_mean                                              -0.0240         -0.0243            85            83            85            83
fizzy/instantiate/sha1_mean                                        -0.0138         -0.0136            91            89            91            89
fizzy/execute/sha1/512_bytes_rounds_1_mean                         -0.0327         -0.0330           344           332           344           332
fizzy/execute/sha1/512_bytes_rounds_16_mean                        -0.0166         -0.0165          4745          4666          4744          4666
fizzy/parse/sha256_mean                                            -0.0109         -0.0111           140           138           140           138
fizzy/instantiate/sha256_mean                                      -0.0144         -0.0142           145           143           145           143
fizzy/execute/sha256/512_bytes_rounds_1_mean                       -0.0018         -0.0020           378           377           378           377
fizzy/execute/sha256/512_bytes_rounds_16_mean                      -0.0033         -0.0030          5306          5288          5304          5288
fizzy/parse/taylor_pi_mean                                         +0.0398         +0.0396             5             6             5             6
fizzy/instantiate/taylor_pi_mean                                   +0.0259         +0.0261            11            11            11            11
fizzy/execute/taylor_pi/pi_1000000_runs_mean                       +0.0807         +0.0805        123672        133650        123662        133615
fizzy/parse/micro/eli_interpreter_mean                             +0.0159         +0.0162             8             8             8             8
fizzy/instantiate/micro/eli_interpreter_mean                       +0.0258         +0.0255            14            14            14            14
fizzy/execute/micro/eli_interpreter/exec105_mean                   +0.0065         +0.0067            15            15            15            15
fizzy/parse/micro/factorial_mean                                   +0.1297         +0.1293             2             3             2             3
fizzy/instantiate/micro/factorial_mean                             +0.1257         +0.1259             2             3             2             3
fizzy/execute/micro/factorial/20_mean                              -0.1138         -0.1139             1             1             1             1
fizzy/parse/micro/fibonacci_mean                                   +0.1071         +0.1068             3             3             3             3
fizzy/instantiate/micro/fibonacci_mean                             +0.1030         +0.1033             3             3             3             3
fizzy/execute/micro/fibonacci/24_mean                              -0.0466         -0.0468         13483         12855         13482         12851
fizzy/parse/micro/host_adler32_mean                                +0.0731         +0.0734             3             4             3             4
fizzy/instantiate/micro/host_adler32_mean                          +0.0435         +0.0432             6             7             6             7
fizzy/execute/micro/host_adler32/1_mean                            +0.0012         +0.0014             0             0             0             0
fizzy/execute/micro/host_adler32/1000_mean                         +0.0033         +0.0031            60            60            60            60
fizzy/parse/micro/spinner_mean                                     +0.1335         +0.1337             2             2             2             2
fizzy/instantiate/micro/spinner_mean                               +0.1166         +0.1164             2             3             2             3
fizzy/execute/micro/spinner/1_mean                                 +0.0139         +0.0142             0             0             0             0
fizzy/execute/micro/spinner/1000_mean                              -0.0081         -0.0084            18            18            18            18
fizzy/parse/stress/guido-fuzzer-find-1_mean                        +0.0418         +0.0420           227           237           227           237
fizzy/instantiate/stress/guido-fuzzer-find-1_mean                  +0.0450         +0.0446           274           286           274           286
```

File: lib/fizzy/execute.cpp [hunk 6]
```diff
@@ -634,12 +625,14 @@ ExecutionResult execute(Instance& instance, FuncIdx func_idx, const Value* args,
                 goto trap;
 
             // check actual type against expected type
-            const auto& actual_type = called_func->type;
```
Comment (axic) [inline lib/fizzy/execute.cpp:637]: Why is this not working anymore when the `called_func` declaration above hasn't changed?

**Comment by gumb0** [comment]
```
fizzy/parse/blake2b_mean                                           +0.0056         +0.0062            44            44            44            44
fizzy/instantiate/blake2b_mean                                     -0.0057         -0.0042            51            51            51            51
fizzy/execute/blake2b/512_bytes_rounds_1_mean                      -0.0859         -0.0850           175           160           175           160
fizzy/execute/blake2b/512_bytes_rounds_16_mean                     -0.0708         -0.0701          2583          2400          2581          2400
fizzy/parse/ecpairing_mean                                         +0.0004         +0.0010          2590          2591          2588          2591
fizzy/instantiate/ecpairing_mean                                   +0.0009         +0.0013          2677          2680          2676          2680
fizzy/execute/ecpairing/onepoint_mean                              -0.1139         -0.1136        889870        788484        889503        788470
fizzy/parse/keccak256_mean                                         -0.0044         -0.0045            81            81            81            81
fizzy/instantiate/keccak256_mean                                   +0.0025         -0.0028            88            88            89            88
fizzy/execute/keccak256/512_bytes_rounds_1_mean                    -0.0418         -0.0455           220           211           221           211
fizzy/execute/keccak256/512_bytes_rounds_16_mean                   -0.0369         -0.0394          3198          3079          3206          3079
fizzy/parse/memset_mean                                            +0.0123         +0.0105            11            11            11            11
fizzy/instantiate/memset_mean                                      +0.0330         +0.0317            17            17            17            17
fizzy/execute/memset/256_bytes_mean                                -0.0757         -0.0766            14            13            14            13
fizzy/execute/memset/60000_bytes_mean                              -0.0776         -0.0783          3083          2844          3085          2843
fizzy/parse/mul256_opt0_mean                                       -0.0054         -0.0061            14            14            14            14
fizzy/instantiate/mul256_opt0_mean                                 +0.0175         +0.0169            20            20            20            20
fizzy/execute/mul256_opt0/input1_mean                              -0.1371         -0.1328            64            55            64            55
fizzy/parse/ramanujan_pi_mean                                      -0.0098         -0.0065            47            46            46            46
fizzy/instantiate/ramanujan_pi_mean                                -0.0046         -0.0024            53            52            53            52
fizzy/execute/ramanujan_pi/33_runs_mean                            -0.0358         -0.0344           230           222           230           222
fizzy/parse/sha1_mean                                              +0.0038         +0.0048            73            73            73            73
fizzy/instantiate/sha1_mean                                        +0.0061         +0.0069            79            80            79            80
fizzy/execute/sha1/512_bytes_rounds_1_mean                         -0.1249         -0.1245           197           173           197           173
fizzy/execute/sha1/512_bytes_rounds_16_mean                        -0.1467         -0.1463          2801          2390          2800          2390
fizzy/parse/sha256_mean                                            +0.0046         +0.0050           124           124           124           124
fizzy/instantiate/sha256_mean                                      +0.0016         +0.0011           131           131           131           131
fizzy/execute/sha256/512_bytes_rounds_1_mean                       -0.0004         -0.0007           175           175           175           175
fizzy/execute/sha256/512_bytes_rounds_16_mean                      -0.0297         -0.0299          2465          2392          2465          2392
fizzy/parse/taylor_pi_mean                                         +0.0185         +0.0183             5             5             5             5
fizzy/instantiate/taylor_pi_mean                                   +0.0644         +0.0643            11            11            11            11
fizzy/execute/taylor_pi/pi_1000000_runs_mean                       -0.0234         -0.0235         87590         85538         87587         85532
fizzy/parse/micro/eli_interpreter_mean                             -0.0024         -0.0024             8             7             8             7
fizzy/instantiate/micro/eli_interpreter_mean                       +0.0172         +0.0173            14            14            14            14
fizzy/execute/micro/eli_interpreter/exec105_mean                   -0.1043         -0.1043            10             9            10             9
fizzy/parse/micro/factorial_mean                                   +0.0040         +0.0040             2             2             2             2
fizzy/instantiate/micro/factorial_mean                             +0.0106         +0.0106             2             2             2             2
fizzy/execute/micro/factorial/20_mean                              -0.0507         -0.0507             1             1             1             1
fizzy/parse/micro/fibonacci_mean                                   +0.0188         +0.0188             2             2             2             2
fizzy/instantiate/micro/fibonacci_mean                             +0.0020         +0.0021             3             3             3             3
fizzy/execute/micro/fibonacci/24_mean                              -0.0139         -0.0139         11010         10856         11008         10855
fizzy/parse/micro/host_adler32_mean                                +0.0326         +0.0327             3             3             3             3
fizzy/instantiate/micro/host_adler32_mean                          +0.0118         +0.0119             7             7             7             7
fizzy/execute/micro/host_adler32/1_mean                            -0.0197         -0.0198             0             0             0             0
fizzy/execute/micro/host_adler32/1000_mean                         -0.0720         -0.0723            68            63            68            63
fizzy/parse/micro/icall_hash_mean                                  +0.0147         +0.0144             6             6             6             6
fizzy/instantiate/micro/icall_hash_mean                            -0.0153         -0.0156            13            12            13            12
fizzy/execute/micro/icall_hash/1000_steps_mean                     -0.3250         -0.3252           207           140           207           140
fizzy/parse/micro/spinner_mean                                     +0.0220         +0.0218             2             2             2             2
fizzy/instantiate/micro/spinner_mean                               +0.0243         +0.0241             2             2             2             2
fizzy/execute/micro/spinner/1_mean                                 -0.0229         -0.0231             0             0             0             0
fizzy/execute/micro/spinner/1000_mean                              -0.1495         -0.1497            19            16            19            16
fizzy/parse/stress/guido-fuzzer-find-1_mean                        -0.0069         -0.0070           267           265           267           265
fizzy/instantiate/stress/guido-fuzzer-find-1_mean                  +0.0016         +0.0015           311           311           311           311
```

**Review by axic — APPROVED** [review]

**Pull request merged**